packages/sort-jsonc/src/sortJsonc.ts

```diff
@@ -79,7 +105,13 @@ export function sortDeepWithSymbols<T extends Record<string | symbol, any>>(init
     }
 
     if (!Array.isArray(current)) {
-      keys.sort(compareFn);
+      alreadySorted = isSortedArray(keys, compareFn);
+      if (checkOnly) {
+        return alreadySorted;
+      }
+      if (!alreadySorted) {
+        keys.sort(compareFn);
+      }
```

Comment:
> If I read this correctly, isSortedArray(keys, compareFn) only checks that the keys of the current object are sorted, and with checkOnly we always return on the first iteration, so it ends up only checking that the first level is sorted.
>
> This can be verified by moving around the nested object keys in the "sorted input returns true"-test.

D'oh! Of course! I should have been more careful with the test cases.

Reply:
Committed the change you suggested but I see what you're saying about performance.